app/scripts/controllers/network/createInfuraClient.js

```diff
@@ -11,7 +11,12 @@ import BlockTracker from 'eth-block-tracker'
 import * as networkEnums from './enums'
 
 export default function createInfuraClient ({ network }) {
-  const infuraMiddleware = createInfuraMiddleware({ network, maxAttempts: 5, source: 'metamask' })
+  const infuraMiddleware = createInfuraMiddleware({
+    network,
+    maxAttempts: 5,
+    source: 'metamask',
+    dev: process.env.NODE_ENV === 'development'
```

Comment:
This gets explicitly set during build, and will reliably either be `'development'` or `'production'`